Add UI for icon scaling options (rel. to #14388)

moving-bits opened the pull request:
## Description
Adds settings' UI for icon scaling options as discussed in https://github.com/cgeo/cgeo/pull/14388#issuecomment-1622076437

![image](https://github.com/cgeo/cgeo/assets/3754370/25799cd1-8a33-49a8-bf56-66ade16d39d0)

Two caveats:
- I needed to rename the preference key, as seekbar preferences require an `int` preference, but for some reason I had created the scaling factors as `long` values. This will reset icon scaling factors once on upgrade.
- ~Currently a restart of c:geo is required to bring changed settings into effect.~

~I've tried resetting the marker cache + updating the scaling factor variables in `MapMarkerUtils` on changing the scaling settings' values, but on opening the map the next time within the same session, no markers got drawn and the map crashed eventually, without any useful info in Logcat. Any ideas?~

moving-bits commented:
> markers are cached in overlaysCache in MapMarkerUtils, you'd have to clear that cache

thanks. I had tried clearing overlayCache and emojiPaintMap + setting scalingFactorCacheIcons /scalingFactorWpIcons to their new values, but on opening the map the next time, it stayed with the cache loading indicator, did not paint any icons/markers, and eventually crashed (without any c:geo related error in the logs).

ztNFny commented:
we have the same issue (require restart) for large smileys btw. Never tried making it work without restart

ztNFny commented:
> I could actually live with the restart - changing the scaling options is probably nothing you would do on a daily basis - but I would like to understand why clearing the overlayCache isn't enough in this case. I guess I must have missed something in my code.

it works for me - I did a quick test on top of my current dev branch. I added the reset call on "offline log" because that's what I was working on and it gets me real-time results
https://github.com/ztNFny/cgeo/commit/e006b3c39d9f6fba6c4f97cc3299a4065a25f186

after logging a cache it's redrawn with its new size:
![image](https://github.com/cgeo/cgeo/assets/1258173/a8faa0f2-3580-4f0b-9f70-1ce7f92bf051)

after going back to list and to map again all markers are the new size

moving-bits commented:
tried this on a different emulator, with the same result. But finally got some logging info:
`OOM allocating Bitmap with dimensions 14250 x 14250`
Image dimensions are way too large.
On assigning the new value manually in `MapMarkerUtils` I need to divide by 100.
Oops.
I'll adapt this PR accordingly.

ztNFny commented:
public synchronized static void resetCache() {
        overlaysCache.clear();
        emojiPaintMap.clear();
        scalingFactorCacheIcons = Settings.getLong(R.string.pref_mapCacheScale, 100) / 100.0f;
        scalingFactorWpIcons = Settings.getLong(R.string.pref_mapWpScale, 100) / 100.0f;
    }


should probably be an init method to make sure init and reset do the same thing

moving-bits commented:
updated PR:
- scaling gets applied on opening the map, no restart of c:geo required anymore
- `resetAllCaches()` is called during class init
- found and fixed a value mapping error in `SeekBarPreference` in conjunction with `onChangePreferenceListener`
